src/main/scala/com/qubole/spark/hiveacid/reader/v2/HiveAcidInputPartitionReaderV2.scala

```
@@ -0,0 +1,88 @@
+  private val jobConf = new JobConf(broadcastedConf.value.value)
+  private val orcColumnarBatchReader = new OrcColumnarBatchReader(1024)
+
+  private def initReader() : Unit = {
+    // Get the reader schema using the column names and types set in hive conf.
+    val readerSchema: TypeDescription =
+      com.qubole.shaded.hadoop.hive.ql.io.orc.OrcInputFormat.getDesiredRowTypeDescr(jobConf, true, 2147483647)
+
+    // Set it as orc.mapred.input.schema so that the reader will read only the required columns
+    jobConf.set("orc.mapred.input.schema", readerSchema.toString)
+
+    val fileSplit = split.inputSplit.value.asInstanceOf[OrcSplit]
+    val readerLocal = OrcFile.createReader(fileSplit.getPath,
+      OrcFile.readerOptions(jobConf).maxLength(
+        OrcConf.MAX_FILE_LENGTH.getLong(jobConf)).filesystem(fileSplit.getPath.getFileSystem(jobConf)))
+
+    // Get the column id from hive conf para. TODO : Can be sent via a parameter
+    val colIds = jobConf.get(ColumnProjectionUtils.READ_COLUMN_IDS_CONF_STR)
+    val requestedColIds = if (!colIds.isEmpty()) {
+      colIds.split(",").map(a => a.toInt)
+    } else {
+      Array[Int]()
+    }
+
+    // Register the listener for closing the reader before init is done.
+    val attemptId = new TaskAttemptID(new TaskID(new JobID(), TaskType.MAP, 0), 0)
+    val taskAttemptContext = new org.apache.hadoop.mapred.TaskAttemptContextImpl(jobConf, attemptId)
+    val iter = new org.apache.spark.sql.execution.datasources.RecordReaderIterator(orcColumnarBatchReader)
+    Option(TaskContext.get()).foreach(_.addTaskCompletionListener[Unit](_ => iter.close()))
+
+    //TODO: Need to generalize it for supporting other kind of file format.
+    orcColumnarBatchReader.initialize(fileSplit, taskAttemptContext)
+    orcColumnarBatchReader.initBatch(readerLocal.getSchema, requestedColIds,
+      requiredFields, partitionSchema, partitionValues, isFullAcidTable && !fileSplit.isOriginal)
```

Shouldn't we be creating orcColumnarBatchReader only if these conditions are met: 
``` isFullAcidTable && !fileSplit.isOriginal```

Right now `orcColumnarBatchReader` would work for both acid and non acid reads because we are creating a baseRecordReader but creating AcidRecordReader only if fullAcidScan is true. 

Here we should only create vectorized reader for acid reads. For normal reads, we should simply throw an exception.

---

fullAcidScan is for full ACID table. The base reader will be used for insert-only table. In that case there will not be any filtering from Hive ACID reader as it does not support update/delete. I think spark native reader does not support read from insert-only Hive ACID tables. If the user has a ACID table with old file format(original), then HiveAcid reader should be able to read it. I don't think we should throw exception.